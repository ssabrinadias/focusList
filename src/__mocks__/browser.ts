import { http, HttpResponse } from "msw";
import { setupWorker } from "msw/browser";

// import { handlers } from "./handlers";

export const worker = setupWorker(
  http.get("...", () => HttpResponse.json({ id: "abc-123" }))
);
console.log(worker.listHandlers());
