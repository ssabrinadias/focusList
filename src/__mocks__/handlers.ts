import { rest } from "msw";

export const handlers = [
  rest.get(
    "https://6664c2d2932baf9032ac056c.mockapi.io/tasks",
    (_, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json([
          {
            createdAt: "2024-06-09T02:50:41.047Z",
            title: "title 1",
            description: "description 1",
            status: "done",
            active: false,
            id: "1",
          },
          {
            createdAt: "2024-06-08T06:47:05.841Z",
            title: "title 2",
            description: "description 2",
            status: "doing",
            active: false,
            id: "2",
          },
          {
            createdAt: "2024-06-08T12:24:47.637Z",
            title: "title 3",
            description: "description 3",
            status: "doing",
            active: false,
            id: "3",
          },
          {
            createdAt: "2024-06-08T02:58:00.152Z",
            title: "title 4",
            description: "description 4",
            status: "done",
            active: false,
            id: "4",
          },
        ])
      );
    }
  ),
];
