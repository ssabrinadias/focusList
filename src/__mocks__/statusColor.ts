export const getTaskStatusColor = jest.fn((status: string) => {
  if (status === "done") return "primary";
  return "disabled";
});
