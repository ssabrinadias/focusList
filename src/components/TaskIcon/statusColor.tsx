const statusColors: Record<string, ValidColorKeys> = {
  todo: "primary",
  done: "disabled",
};

type ValidColorKeys =
  | "disabled"
  | "action"
  | "inherit"
  | "primary"
  | "secondary"
  | "error"
  | undefined;

export const getTaskStatusColor = (status: string): ValidColorKeys => {
  return statusColors[status] ?? "primary";
};
