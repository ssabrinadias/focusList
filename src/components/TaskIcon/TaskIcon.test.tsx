import { render } from "@testing-library/react";

import { getTaskStatusColor } from "../../__mocks__/statusColor";

import TaskStatusIcon from "./index";

jest.mock("../../__mocks__/statusColor");

describe("TaskStatusIcon Component", () => {
  test("renders with primary color when status is done", () => {
    (getTaskStatusColor as jest.Mock).mockReturnValue("primary");

    render(<TaskStatusIcon status="done" />);
    expect(screen).toMatchSnapshot();
  });

  test("renders with disabled color when status is todo", () => {
    (getTaskStatusColor as jest.Mock).mockReturnValue("disabled");

    render(<TaskStatusIcon status="todo" />);
    expect(screen).toMatchSnapshot();
  });
});
