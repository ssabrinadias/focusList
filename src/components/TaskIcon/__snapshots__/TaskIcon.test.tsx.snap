// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskStatusIcon Component renders with disabled color when status is todo 1`] = `Screen {}`;

exports[`TaskStatusIcon Component renders with primary color when status is done 1`] = `Screen {}`;
